src/main/java/com/ibm/watson/modelmesh/ModelMeshApi.java

```diff
@@ -300,6 +297,13 @@ protected void adjustThreadCount(int delta) {
     }
 
     public void shutdown(long timeout, TimeUnit unit) throws InterruptedException {
+        if (payloadProcessor != null) {
+            try {
+                payloadProcessor.close();
+            } catch (IOException e) {
+                logger.warn("Error closing PayloadProcessor {}: {}", payloadProcessor, e.getMessage());
+            }
+        }
```

Comment:
Would be best to move this a bit further down, maybe after the if block below. To minimize the chance of requests being missed (some requests may continue to arrive during the shutdown process).

Reply:
ok, will do